### PR TITLE
Ftr/8962 status of funds chart table toggle

### PR DESCRIPTION
**Pull request by brianpetway**
**High level description:**

This pr is for tickets 8962, 8965, and 8939 (links below).
We added a toggle button component to switch between the chart and a table view in the Status of Funds section on the Agency page. And we used the dtui Table component with its responsive version, using the 'isStacked' flag.
I had to re-arrange the components in the header of this section because the design has the new toggle in a different place in the DOM for mobile than it is for desktop and while doing that I fixed the existing issue of the 'What is This?' accordion's caret floating around as you change screen sizes.
I also made a new dtui release that contains the responsive css for the Table component (that work was done by Stephanie from Design).
Also, the icon used for 'chart' is different from the one in the mock. The one used here is one that has been used elsewhere on the site and the one in the mocks is not in the free fontawesome library that we use.

**Technical details:**

see above

**JIRA Ticket:**
[DEV-8962](https://federal-spending-transparency.atlassian.net/browse/DEV-8962)
[DEV-8965](https://federal-spending-transparency.atlassian.net/browse/DEV-8965)
[DEV-8939](https://federal-spending-transparency.atlassian.net/browse/DEV-8939)

**Mockup:**
mobile - https://app.zeplin.io/project/61f806d9ad825610f7b1fc0a?seid=6282cb9cd1398515c75ddb7b
desktop - https://app.zeplin.io/project/61f806d9ad825610f7b1fc0a?seid=6256d35e2e10951f44514621

The following are ALL required for the PR to be merged:

Author:
- [ x] Linked to this PR in JIRA ticket
- [ ] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [ x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Edge
- [ x] Verified mobile/tablet/desktop/monitor responsiveness
- [ ] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [ ] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
- [ ] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [ ] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [ ] Design review complete `if applicable`
- [ ] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete

### CODE REVIEW

File: src/_scss/pages/agency/index.scss [hunk 1]
```diff
@@ -29,146 +29,65 @@
...
-            float:none;
-            width:33%;
-            margin-left: -8px;
-        }
-        @media (min-width: 320px) and (max-width: 400px) {
-            float:none;
-            width:36.9%;
-            margin-left: -8px;
-        }
+        font-size: rem(14);
```
Comment (nick-torres) [inline src/_scss/pages/agency/index.scss:42]: should probably start using the new vars stephanie made $font-size-14

File: src/_scss/pages/agency/index.scss [hunk 2]
```diff
@@ -29,146 +29,65 @@
...
     }
 
     .accordion-container .toggle {
-        @media(max-width: $tablet-screen) {
-            margin-right: 32px;
-            margin-left: 4px;
-            padding: 0;
-        }
         color: $color-primary;
+        font-size: rem(14);
```
Comment (nick-torres) [inline src/_scss/pages/agency/index.scss:75]: see above

File: src/js/components/agency/statusOfFunds/VisualizationSection.jsx [hunk 3]
```diff
@@ -52,22 +57,132 @@ const VisualizationSection = ({
...
-                className="status-of-funds__controls">
-                <RoundedToggle toggle={toggle} onKeyToggle={onKeyToggle} onToggle={onToggle} label="View Outlays" />
-                <div className="status-of-funds__line-div" />
-                <div className="status-of-funds__accordion">
-                    <Accordion setOpen={setOpen} closedIcon="chevron-down" openIcon="chevron-up" title="What is this?" />
-                </div>
-            </div>
+            {isMobile ? (
+                <>
+                    <h6>{level === 1 ? selectedSubcomponent?.name : agencyName} by <span className="status-of-funds__emphasis">{levels[level]}</span> for FY {fy}
```
Comment (nick-torres) [inline src/js/components/agency/statusOfFunds/VisualizationSection.jsx:158]: should you be using a different heading tag and then just making the font size the size of the h6 tag?

File: src/js/components/agency/statusOfFunds/VisualizationSection.jsx [hunk 4]
```diff
@@ -52,22 +57,132 @@ const VisualizationSection = ({
...
+                        <Accordion setOpen={setOpen} closedIcon="chevron-down" openIcon="chevron-up" title={accordionTitle} />
+                    </div>
+                </>
+            )
+                :
+                (
+                    <>
+                        <div className="status-of-funds__controls">
+                            <div className="status-of-funds__controls-desktop-row-one">
+                                <h6>{level === 1 ? selectedSubcomponent?.name : agencyName} by <span className="status-of-funds__emphasis">{levels[level]}</span> for FY {fy}
```
Comment (nick-torres) [inline src/js/components/agency/statusOfFunds/VisualizationSection.jsx:174]: see above

Comment (brianpetway) [reply src/js/components/agency/statusOfFunds/VisualizationSection.jsx:174]: we decided it's better to leave these changes for another ticket which will address more of these type of issues all at once

**Comment by nick-torres** [comment]
in the mock the toggle looks higher than in your PR
https://app.zeplin.io/project/61f806d9ad825610f7b1fc0a/screen/6279596279254e12040a52af
image

**Comment by nick-torres** [comment]
mobile looks pretty close

**Review by nick-torres — APPROVED** [review]

**Pull request merged**